src/Core.tsx

```diff
@@ -407,7 +405,8 @@ export class Core extends React.Component<IModalProps, IModalState> {
    }
 
    private onConfirmSelectiveDisclosure (sd: SD) {
-     this.setState({ sd, currentStep: 'confirmInformation' })
+     this.setState({ sd })
+     this.shouldShowConfirmStep()
```

Comment:
I am getting error from the backend because it is not receiving the `sd`

<img width="487" alt="Screen Shot 2022-03-28 at 13 29 32" src="https://user-images.githubusercontent.com/36084092/160446487-56448e41-c16e-4191-8012-601205c93a8a.png">

![Screen Shot 2022-03-28 at 13 41 47](https://user-images.githubusercontent.com/36084092/160446462-84423ae3-98f5-48f9-a810-f268029fcf20.png)

Adding log in line 414 showed me that `sd` is not set in the state. I tried with this change and it worked fine

```suggestion
     this.setState({ sd }, () => this.shouldShowConfirmStep())
```

So now `shouldShowConfirmStep` is called after state has been updated.

Reply:
Additional, we could add an integration test intercepting what is sent to the backend no?

Reply:
Great! But then it is not sent to the backend after getting them

Reply:
Okay! [Another test for this written here](https://github.com/rsksmart/rLogin/pull/242/files#diff-62b4d7c28e6178f6a760978353e1ef2d4075429279da31a5dbb2f40351ab53a2R285) with the fix suggested above. This passes!